Brigade worker build error

I seem to be hitting some interesting build errors for the Brigade worker (both on Linux (WSL) and Windows):

```
$ tsc
node_modules/@kubernetes/client-node/dist/exec.d.ts(2,23): error TS2688: Cannot find type definition file for 'ws'.
node_modules/@kubernetes/client-node/dist/portforward.d.ts(2,23): error TS2688: Cannot find type definition file for 'ws'.
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts(2,23): error TS2688: Cannot find type definition file for 'ws'.
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts(13,35): error TS1039: Initializers are not allowed in ambient contexts.
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts(14,36): error TS1039: Initializers are not allowed in ambient contexts.
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts(15,36): error TS1039: Initializers are not allowed in ambient contexts.
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts(16,36): error TS1039: Initializers are not allowed in ambient contexts.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Investigation in progress.

Adding `@types/ws` solves part of the problem - still hitting:

```
$ tsc







14     static readonly StdoutStream = 1;
                                      ~

node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:15:36 - error TS1039: Initializers are not allowed in ambient contexts.

15     static readonly StderrStream = 2;
                                      ~

node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:16:36 - error TS1039: Initializers are not allowed in ambient contexts.

16     static readonly StatusStream = 3;
```

Added ` @types/ws @types/node @types/request`, still hitting:

```
$ tsc
node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:13:35 - error TS1039: Initializers are not allowed in ambient contexts.

13     static readonly StdinStream = 0;
                                     ~

node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:14:36 - error TS1039: Initializers are not allowed in ambient contexts.

14     static readonly StdoutStream = 1;
                                      ~

node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:15:36 - error TS1039: Initializers are not allowed in ambient contexts.

15     static readonly StderrStream = 2;
                                      ~

node_modules/@kubernetes/client-node/dist/web-socket-handler.d.ts:16:36 - error TS1039: Initializers are not allowed in ambient contexts.

16     static readonly StatusStream = 3;
```

--- a/Makefile
+++ b/Makefile
@@ -103,7 +103,7 @@ test-functional:
 # JS test is local only
 .PHONY: test-js
 test-js:
-	cd brigade-worker && KUBECONFIG="./test/fake_kubeconfig.yaml" yarn test
+	cd brigade-worker && yarn build && KUBECONFIG="./test/fake_kubeconfig.yaml" yarn test
 
 .PHONY: test-style
 test-style:

--- a/brigade-worker/package.json
+++ b/brigade-worker/package.json
@@ -28,7 +28,8 @@
     "rimraf": "^2.6.2",
     "ts-node": "^7.0.0",
     "typedoc": "^0.8.0",
-    "typescript": "^3.0.1"
+    "typescript": "^3.2.2",
+    "@types/ws": "6.0.1"
   },
   "dependencies": {
     "@kubernetes/client-node": "^0.7.2",

--- a/brigade-worker/test/mock.ts
+++ b/brigade-worker/test/mock.ts
@@ -66,7 +66,8 @@ export class MockJob extends Job {
         }, delay);
         return;
       }
-      setTimeout(resolve(new MockResult(this.name)), delay);
+      resolve(new MockResult(this.name))
+      setTimeout(null, delay)
     });
   }
   public logs(): Promise<string> {

